### PR TITLE
HGC TPG validation improvement

### PR DESCRIPTION
**Pull request by ebecheva**
Some new features were added in order to :
- customize the validation : possibility to run separately ref and test validations, or choose between the different parts of the simulation (install, compile, run), to configure environment depending on the site (LLR or CERN)  

- check the progress of the validation process and displaye on the screen and in the log file

### CODE REVIEW

File: config/user_parameters_cfg.py [hunk 1]
```diff
@@ -5,6 +5,12 @@
 # Install working area for reference release and/or branch and
 # Generate configuration file that will simulate tpgs
 generate_configFileParameters = ConfigFileParameters(
+    validationRef = 'yes', #yes or no
+    validationTest = 'yes', #yes or no
+    installStep = 'yes', #yes or no
+    compileStep = 'yes', #yes or no
+    simulationStep = 'yes', #yes or no
```
Comment (jbsauvan) [inline config/user_parameters_cfg.py]: Would be better to have booleans instead of `yes`/`no`

File: hgctpgvalidation/parameters.py [hunk 2]
```diff
@@ -26,44 +32,92 @@ class ConfigFileParameters():
...
         'git checkout -b ' + self.localTestBranchName + ' ' + self.remoteTest + '/' + self.remoteTestBranchName + ';' + \
-        'scram b -j4; ' + 'echo === End of compilation ===;' + 'echo $PWD;' + \
-        'cmsDriver.py hgcal_tpg_validation -n ' + str(self.numberOfEvents) + ' --mc  --eventcontent FEVTDEBUG --datatier GEN-SIM-DIGI-RAW --conditions ' + self.conditions + ' ' + \
-        '--beamspot ' + self.beamspot + ' ' + '--step ' + self.step + ' ' + \
-        '--geometry ' + self.geometryTest + ' ' + '--era ' + self.eraTestName + ' ' + '--procModifiers ' + self.procModifiers + ' ' + \
-        '--inputCommands "keep *",' + self.dropedBranches + ' ' + \
-        '--filein ' + self.inputTestFileName + ' ' + \
-        '--no_output ' + '--customise=' + self.customiseTestFile + ' ' + \
-        '--customise_commands "process.schedule = cms.Schedule(process.user_step)"' + ';'
+        'scram b -j4; ' + 'echo === End of compilation ===;' + 'echo $PWD;'
```
Comment (jbsauvan) [inline hgctpgvalidation/parameters.py]: The compilation should not be there since it is in `runCompileTestStep`

File: scripts/validation_tpg.py [hunk 3]
```diff
@@ -42,27 +53,133 @@ def launchPlotHistos(parameters, logfile):
...
-    subproc = subprocess.Popen(sourceCmd, stdout=logfile, stderr=logfile)
-    subproc.communicate()
-    print (logfile, 'Finished plot')
+    logfile = open('logfile', 'a+')
+    subProc = subprocess.Popen(sourceCmd, stdout=logfile, stderr=logfile)
+    subProc.communicate()
+    checkSubprocessStatus(subProc, logfile)
+    print('Plots were created.')
+    #logfile = open('logfile', 'a+')
+    #logfile.write('Finished creating plots.')
```
Comment (jbsauvan) [inline scripts/validation_tpg.py]: Code commented out to be removed

**Comment by ebecheva** [comment]
Some additional corrections for HGC TPG validation package

**Comment by ebecheva** [comment]
Some additional corrections for HGC TPG validation package

**Pull request merged**